plugins/GSdx/GSRendererHW.cpp

```diff
@@ -841,6 +841,25 @@ void GSRendererHW::Draw()
 			m_vt.m_max.t = tmax;
 		}
 
+		// Code for Automatic Mipmapping. Relies on game CRCs.
+		switch (m_game.title) {
+			// Games requiring basic mipmapping
+		case CRC::AceCombat4:
+		case CRC::AceCombat5:
+		case CRC::RatchetAndClank:
+		case CRC::RatchetAndClank2:
+		case CRC::RatchetAndClank3:
+		case CRC::RatchetAndClank4:
+		case CRC::RatchetAndClank5:
+			auto_mipmap_basic = m_mipmap == -1;
+				break;
+			// Games requiring full mipmapping
+		case CRC::ApeEscape2:
+			auto_mipmap_full = m_mipmap == -1;
+				break;
+		}
```

Comment:
I'm not sure if this really fits in this section; and I'm sure there's a more efficient way to do this without a bunch of case labels.

Reply:
Shot in the dark; if there is a register that changes value when a game uses mipmaps, that could be very useful. If such an obscure thing exists.

Reply:
A register exists but I can tell you that nearly all decent 3D games use mipmap. However you don't need to emulate it.

Reply:
Yes it is bad to update this code here. It should be done in the CRC init of GSState. And IsMipMapActive should return true/false.